src/execution/sql/functions/string_functions.cpp

```diff
@@ -370,4 +370,73 @@ void StringFunctions::Like(BoolVal *result, UNUSED_ATTRIBUTE exec::ExecutionCont
   result->val_ = sql::Like{}(string.val_, pattern.val_);  // NOLINT
 }
 
+void StringFunctions::Position(exec::ExecutionContext *ctx, Integer *pos, const StringVal &search_str,
+                               const StringVal &search_sub_str) {
+  if (search_str.is_null_ || search_sub_str.is_null_) {
+    *pos = Integer::Null();
+    return;
+  }
+
+  auto search_str_view = search_str.StringView();
+  auto search_sub_str_view = search_sub_str.StringView();
+
+  // Postgres performs a case insensitive search for Position()
+  auto it =
+      std::search(search_str_view.begin(), search_str_view.end(), search_sub_str_view.begin(),
+                  search_sub_str_view.end(), [](char ch1, char ch2) { return std::toupper(ch1) == std::toupper(ch2); });
+  auto found = (it - search_str_view.begin());
+
+  if (static_cast<size_t>(found) == search_str_view.length()) {
+    *pos = Integer(0);
+  } else {
+    *pos = Integer(found + 1);
+  }
+}
+
+void StringFunctions::ASCII(exec::ExecutionContext *ctx, Integer *result, const StringVal &str) {
+  if (str.is_null_) {
+    *result = Integer::Null();
+    return;
+  }
+
+  if (str.GetLength() == 0) {
+    *result = Integer(0);
+    return;
+  }
+
+  auto str_view = str.StringView();
+  *result = Integer(static_cast<int>(str_view.front()));
+}
+
+void StringFunctions::Chr(exec::ExecutionContext *ctx, StringVal *result, const Integer &code) {
+  auto num = static_cast<uint64_t>(code.val_);
+  if (num == 0 || num > 0x10FFFF) {
+    // FIXME: this should properly throw a function error
```

Comment:
This comment is from the original PR. I don't see any other functions in this class throwing an error, instead they all return Null on errors. I'm going to remove this comment.